neuror/sanitize.py

```diff
@@ -123,3 +129,85 @@ def fix_non_zero_segments(neuron):
     for section in to_be_deleted:
         neuron.delete_section(section)
     return neuron
+
+
+def annotate_neurolucida(morph_path, checkers=None):
+    """Annotate errors on a morphology in neurolucida format.
+
+    Args:
+        morph_path (str): absolute path to an ascii morphology
+        checkers (dict): dict of checker functons from neurom with function as keys
+            and marker data in a dict as values, if None, default checkers are used
+
+    Default checkers include:
+        - fat ends
+        - z-jumps
+        - narrow start
+        - dangling branch
+        - multifurcation
+
+    Returns:
+        annotations to append to .asc file
+        dict of error summary
+        dict of error markers
+    """
+    if checkers is None:
+        checkers = {
+            nc.has_no_fat_ends: {"name": "fat end", "label": "Circle3", "color": "Blue"},
+            partial(nc.has_no_jumps, axis="z"): {
+                "name": "zjump",
+                "label": "Circle2",
+                "color": "Green",
+            },
+            nc.has_no_narrow_start: {"name": "narrow start", "label": "Circle1", "color": "Blue"},
+            nc.has_no_dangling_branch: {"name": "dangling", "label": "Circle6", "color": "Magenta"},
+            nc.has_multifurcation: {
+                "name": "Multifurcation",
+                "label": "Circle8",
+                "color": "Yellow",
+            },
+        }
+
+    def _try(checker, neuron):
+        """Try to apply a checker, and return False if exception raised"""
+        try:
+            return checker(neuron)
+        except Exception as e:  # pylint: disable=broad-except
+            L.exception("%s failed on %s", checker, morph_path)
+            L.exception(e, exc_info=True)
+            return CheckResult(True)
+
+    neuron = load_neuron(morph_path)
+    results = [_try(checker, neuron) for checker in checkers]
+    markers = [
+        dict(setting, data=result.info)
+        for result, setting in zip(results, checkers.values())
+        if not result.status
+    ]
+    summary = {
+        setting["name"]: len(result.info)
+        for result, setting in zip(results, checkers.values())
+        if result.info
+    }
+    return annotate(results, checkers.values()), summary, markers
```

Comment:
I think it is best to simply return `results` and let the user decide how he wants to use this data. The way the data is currently returned is very specific.

Reply:
but results does not contain enough information, we need to at least return reuslts and checkers, so we know which results is for what checkers. The summary thing was copy pasted from morphservice, I just added the marker one, to have a proper json with all the data, instead of only the .asc annotated morph. I would drop the summary one, as it only contains partial information, but I left it to do as morphservice was doing.